Sources/NIOHTTP2/WatermarkedFlowController.swift

```diff
@@ -62,15 +63,18 @@ struct WatermarkedFlowController {
 extension WatermarkedFlowController {
     /// Notifies the flow controller that we have buffered some bytes to send to the network.
     mutating func bufferedBytes(_ bufferedBytes: Int) {
-        self.pendingBytes += UInt(bufferedBytes)
+        self.pendingBytes += bufferedBytes
         if self.pendingBytes > self.highWatermark {
             self.isWritable = false
         }
     }
 
     /// Notifies the flow controller that we have successfully written some bytes to the network.
     mutating func wroteBytes(_ writtenBytes: Int) {
-        self.pendingBytes -= UInt(writtenBytes)
+        // pendingBytes can become negative if the ordering of calls to wroteBytes(_:) and bufferedBytes(_:)
+        // is incorrect. This can happen if a completed promise is attached to a write in the HTTP2StreamChannel and
+        // the callback to wroteBytes(_:) happens before bufferedBytes(_:).
```

Comment:
Can you elaborate on this a bit to explain why we think it's fine to allow this to happen?

Reply:
Additionally, can you add a `precondition` that `writtenBytes` is positive?